guide/src/reference/cli.md

```diff
@@ -15,23 +15,23 @@ The recommend way to install the `wasm-bindgen` command line tool is with the
 `wasm-pack` installer described
 [here](https://rustwasm.github.io/wasm-pack/installer/). After installing
 `wasm-pack`, you are ready to build project invoking `wasm-pack build`.
-This command installs apropriate version of the `wasm-bindgen` command-line
+This command installs a compatible version of the `wasm-bindgen` command-line
 tool. The version of `wasm-bindgen` installed by `wasm-pack` is not available
- to be used directly via command line.
+to be used directly via command line.
 
-It is not recommended to install `wasm-bindgen-cli` as its version must match
-_exactly_ the version of `wasm-bindgen` that is specified in the project's
-cargo.lock file. Using `wasm-pack` for building simplifies the build process
+It is not recommended to install `wasm-bindgen-cli` as its internal schema
```

Comment:
To be fair, this is much less of an issue now so I think this warning can be lifted especially since the schema hasn't changed in over 6 months. 

But I kept it intact for now in case you still want this here.

Reply:
Actually yeah would you be ok rewording these docs indicating that using the CLI directly is recommended nowadays? Given the current state of things I think it'd be fine to either remove references to `wasm-pack` or mention it only in a subsection below perhaps too

Reply:
Cool, I'll remove all wasm-pack recommendations for build related things.

While we're at it, I think we can replace the [`wasm-pack` test recommendation](https://rustwasm.github.io/wasm-bindgen/wasm-bindgen-test/usage.html#appendix-using-wasm-bindgen-test-without-wasm-pack) with just an example of how to download the webdriver yourself. 

I'll go ahead and do that, but if you happen to see this before I dive in and you think it's a bad idea just let me know.

I'll also include a an example of [setting the runner with an environment variable](https://doc.rust-lang.org/cargo/reference/config.html#targettriplerunner) instead of needing to manage the config file.

Reply:
Now that we're no longer recommending wasm-pack I don't think this quick little hash solution is good enough anymore. A little too brittle. 

For example, if rustfmt got a new rule and we ran `cargo fmt` and the schema file got touched we'd have a new version for no good reason.

I'm going to change the approach to something along the lines of:

1. Generate a `SCHEMA_HASH` in the build script.
2. Set `SCHEMA_VERSION` by hand to the current `Cargo.toml` version
3. Add a test case to the bottom of `shared/src/lib.rs` that asserts that the current hash is equal to the last human approved hash.
4. Whenever the hash changes tests will fail. Then a human can either decide to keep the `SCHEMA_VERSION` the same, or bump it to the latest `Cargo.toml` version if the schema has indeed changed. This would be documented above the test.

---

For my own needs just bumping versions every now and then would've been fine, even if sometimes unnecessary.

I'm only going this far because a lot of people use wasm-bindgen so I don't want the unnecessary churn across the entire ecosystem.

The most robust would probably be to pull the `shared_api!`  macro out into a crate, have the `shared` crate re-export it and have the `shared/build.rs` depend on that `shared_api!` macro directly along with the encode_api! and decode_api! macros and run them all, hash the results and then use that as our source of truth hash.... But I'll hold off on that since the approach suggested above seems like it will meet our needs in practice just fine.

Reply:
For changes to testing, sounds good! For human verification the hash changes, also sounds good!

I'm fine for restructuring things too, but it's probably not too necessary to go too overboard. Honestly things don't change enough nowadays that manually changing the schema version/hash is probably fine.